### ISSUE TITLE
Use a `role="textbox"` for all the Editable elements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#3412]
The "editable" blocks use an element with `contenteditable`. Depending on the cases, this element can be a paragraph, a heading, a div, a figcaption...

Screen readers need to understand an element is an "editable text" element to correctly interact with it. Not surprisingly, browsers expose different information to assistive technologies and in turn assistive technologies have their own quirks.

TL:DR some browser / screen readers combinations don't work well without an explicitly set ARIA role. For example, JAWS 17 with Firefox on Windows doesn't understand a Gutenberg paragraph block is editable and doesn't automatically switch to "forms mode". Users can manually switch to forms mode when they need, and that can be done pressing Enter. However, pressing Enter in Gutenberg creates a new block so this won't work.

The result is the *complete inability to switch to forms mode, so it's impossible to enter content*.

A standard way to let screen readers understand an element has a semantics of editable text is using a `role="textbox"`.

Inputs and textareas don't need any role, since they already have the required semantics.

Multi-line elements should use `role="textbox"` and `aria-multiline="true`.
Single-line elements should use just `role="textbox"`.

To illustrate how browsers differ in exposing information through the accessibility API, see the screenshot below (using the Accessibility Viewer, a tool to explore a11y infos):

## Without role

Firefox exposes a Gutenberg editable paragraph like a paragraph (also notice the weird ARIA role img)

![screenshot 114](https://user-images.githubusercontent.com/1682452/32617085-5b1d73f0-c574-11e7-94c5-1f3503ba2732.png)

IE 11 exposes it as "editable text", this explains why JAWS works nicely with IE and switches to forms mode:

![screenshot 115](https://user-images.githubusercontent.com/1682452/32617154-81c33652-c574-11e7-9e75-f9ef31fc2cd7.png)

## With role=textbox

Firefox:

![screenshot 116](https://user-images.githubusercontent.com/1682452/32617178-9245226a-c574-11e7-85cb-def59be81161.png)

IE 11:

![screenshot 117](https://user-images.githubusercontent.com/1682452/32617179-9261f746-c574-11e7-91b0-12c56ea0582c.png)

both browsers expose now a correct, consistent, information.

**Comment by darrencoutts118** [comment]
+1 this is definitely something that should be added in. Need to be as accessible as possible

**Comment by afercia** [comment]
Detailing here also the issue about labels from https://github.com/WordPress/gutenberg/issues/1659

See screenshot below:

![screenshot 15](https://user-images.githubusercontent.com/1682452/33135827-d4e5986e-cfa3-11e7-8d44-41f412f9543f.png)

The main title is a special case, so let's ignore it for now.

Instead, all the other editable regions use an aria-label attribute `Add text or type...` which is not appropriate. The aria-label attribute works as a `<label>` element and gives a name to its control. So both when the Editable is empty or already filled up with content, screen readers will announce the Editable with the name given by the aria-label.

For example, landing on a paragraphs screen reader users will hear:
`Add text or type / to insert content`

This is OK as a visual instruction, but doesn't tell anything useful to users. In a standard form, this would be the equivalent of, for example, a "first name" field with a label that doesn't say it's about the first name, and says instead `Add text or type / to insert content`.

Instead, the aria-label should identify what the Editable is. For a paragraph, it should say: `Paragraph content` or similar wording.

**Comment by aduth** [comment]
Could this be considered a TinyMCE issue? cc @androb

**Comment by androb** [comment]
@aduth mentioned this issue internally and it appears TimG is on it
